general-concepts/dependencies/text.xml

```diff
@@ -606,6 +615,57 @@ RESTRICT="!test? ( test )"
 DEPEND="test? ( dev-util/foo )"
 </codesample>
 
+</body>
+</section>
+
+<section>
+<title>Circular Dependencies</title>
+<body>
+
+<p>
+Circular dependencies occur if one or more of package's (possibly indirect)
+dependencies depend on the package itself. This creates a dependency cycle where
+each of the packages must technically be installed before the other.
+For example, if package A depends on B, B depends on C and C depends on A, then
+the package manager cannot install A before C, and C before A.
+</p>
+
+<p>
+There are three kinds of circular dependencies:
+</p>
+
+<ol>
+  <li>
+    Weak circular dependencies that occur if only one of the packages strictly
+    needs to be installed before the other. For example,
+    <c>dev-python/certifi</c> strictly requires <c>dev-python/setuptools</c>
+    to build but <c>dev-python/setuptools</c> requires <c>dev-python/certifi</c>
+    only for some runtime functionality, permitting it to be installed later.
+    <c>PDEPEND</c> is used to express this and automatically resolve
+    the circular dependency.
+  </li>
+
+  <li>
+    USE-conditional (non-weak) circular dependencies that occur if the cycle
+    applies only to some combination of USE flags on one of the packages.
+    For example, running tests in <c>dev-python/setuptools</c> requires a number
+    of packages which require <c>dev-python/setuptools</c> to be installed
+    first. This kind of circular dependency can be resolved by the user
+    by adjusting USE flags on one of the packages, e.g. by disabling
+    tests on <c>dev-python/setuptools</c>, and reenabling them once the package
+    is initially installed.
+  </li>
+
+  <li>
+    Strong circular dependencies that occur if cycle cannot be resolved using
+    the regular means. For example, <c>dev-util/cmake</c> used to depend
+    on <c>dev-libs/jsoncpp</c>, while <c>dev-libs/jsoncpp</c> required
+    <c>dev-util/cmake</c> to build. Resolving this kind of dependency usually
+    requires bundling one of the dependencies conditionally, or providing
+    an alternate bootstrap path.
+  </li>
```

Comment:
I agree with gokturk that I also dislike this ternary terminology "weak" vs "non-weak" vs "strong". One would expect that "non-weak" is the opposite of "weak" so there cannot be a third category.

Do we even need names for them? Unless they are referenced from another part of the document, this doesn't seem strictly necessary.

Reply:
The terminology doesn't exist. I don't wanna hold up the PR just for that but I also don't wanna commit to a particular terminology. If we go with "The first kind ...", "The second kind ...", "The third kind ...", I'd be fine with that. Terminology bikeshedding shouldn't hold this up.

Reply:
I think the sentences still make sense even without extra terms like 'first kind'. I'll rewrite them a bit and push the update.